public/app/plugins/datasource/loki/modifyQuery.ts

```diff
@@ -40,15 +42,16 @@ export function addLabelToQuery(query: string, key: string, operator: string, va
   }
 
   const streamSelectorPositions = getStreamSelectorPositions(query);
+  const hasStreamSelectorMatchers = getMatcherInStreamPositions(query).length > 0;
```

Comment:
I am wondering if we should compare the length of matchers with the length of `streamSelectorPositions`. Something like this:

```suggestion
  const hasStreamSelectorMatchers = getMatcherInStreamPositions(query).length >= streamSelectorPositions.length ;
```

The reason might be a super edge case, but I think in binary operations with two queries, e.g. 

```
sum(rate({place="luna"} [$__interval])) + sum(rate({} | json | __error__=`` [$__interval]))
```

we would potentially add the new label to the wrong place. What do you think?

Reply:
I don't think this would solve it because you can have 3 matchers in first stream selector and 0 in next one. But good catch, I am going to look into it.  Just to confirm if we are on the same page, in this case (when you have 1 query with empty selector + parser and 1 with non-empty) we would add adhoc filter to selector at both places, right?

Reply:
I'll try to see how complex it and let you know, if we should do it in separate issue or this one.

Reply:
I'll create a new issue for this, because I wrote some test scenarios where it becomes more complex and I am not sure on how to solve it in those cases. 

We agreed on this `sum(rate({x="y"} [5m])) + sum(rate({} | __error__=`` [5m]))` => `sum(rate({x="y", bar="baz"} [5m])) + sum(rate({bar="baz"}| __error__=`` [5m]))`

But what about `sum(rate({x="y"} [5m])) + sum(rate({x="y"} | __error__=`` [5m]))`? Should we check if each subquery has parser/label filters as well? 

I'll create an issue and we can discuss there what the behaviour should be when you are using adhoc filters for metrics query that has different log queries.